src/service/repo/anonymousMessageRepo.ts

```diff
@@ -138,13 +142,30 @@ async function setChannelAsRead(channelId: string): Promise<boolean> {
   }
 }
 
+async function getAnonymousChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+  try {
+    const response = await anonymousApi.get(
+      baseUrl.getAnonymousChannelDetail(channelType, channelId)
+    );
+    if (response.status === 200) {
+      return Promise.resolve(response.data?.data);
+    }
+
+    return Promise.reject(response.data?.status);
+  } catch (e) {
+    console.log(e);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getAnonymousChannelDetail` function, you are logging the error directly to the console. This might not be ideal for production code as it could potentially expose sensitive information. Consider using a more sophisticated logging mechanism that obscures or removes sensitive data, or even better, send these errors to an error tracking service so they can be monitored and addressed.

Also, when rejecting the promise, it would be better to create an Error object with a message rather than just passing the raw error or response status. This will provide more context about what went wrong if this rejection is caught and logged elsewhere.
```diff
- console.log(e);
- return Promise.reject(e);
+ // Use your preferred logging method here
+ logError(e);
+ return Promise.reject(new Error('Failed to fetch anonymous channel detail'));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->